src/util/ssa_expr.h

```diff
@@ -152,33 +152,34 @@ class ssa_exprt:public symbol_exprt
   }
 };
 
-/// Cast a generic exprt to an \ref ssa_exprt. This is an unchecked conversion.
-/// \a expr must be known to be \ref  ssa_exprt.
+inline bool is_ssa_expr(const exprt &expr)
+{
+  return expr.id()==ID_symbol &&
+         expr.get_bool(ID_C_SSA_symbol);
+}
+
+template <>
+inline bool can_cast_expr<ssa_exprt>(const exprt &base)
+{
+  return is_ssa_expr(base);
+}
```

Comment:
It would be great for this to use `ssa_exprt::check` - see `binary_exprt` in std_expr.h for how I think this could best achieve as much code re-use as possible.

Reply:
They don't appear to use one another? I think this is small enough that it would be simpler for you to do what you want than to explain it to me.

Reply:
Sure, my apologies for taking this long: https://github.com/smowton/cbmc/pull/3